docs/make.jl

```diff
@@ -52,58 +45,167 @@ function generate_example_page(sections)
             ```
             $(cmd_jl_str)
             ```
-            The output will be like this:
-            ```@raw html
-            <div id="$(name)"></div>
-            <script>
-                AsciinemaPlayer.create('assets/$(name).cast', document.getElementById('$(name)'), {
-                    cols: 80,
-                    rows: 24
-                });
-            </script>
-            ```
             """
+
+            if mode == :asciinema
+                path_cast = joinpath(@__DIR__, "src", "assets", "$(name).cast")
+                cmd_record = `asciinema rec $(path_cast) --command $(cmd_jl_str) --overwrite --title $("Replay.jl - "*title)`
+
+                # Dry run not to record precompilation
+                run(cmd_jl)
+                # Record the REPL output with asciinema
+                run(cmd_record)
+
+                # Add the output of the example
+                script_md *= """
+                The output will be like this:
+                ```@raw html
+                <div id="$(name)"></div>
+                <script>
+                    AsciinemaPlayer.create('../assets/$(name).cast', document.getElementById('$(name)'), {
```

Comment:
I understand the situation. On the `main` branch, the parameter `prettyurls` is set to `true` on GitHub Actions while `false` locally. The file named in `example/index.html` is generated on GitHub because `prettyurls` is `true`. 
In this case, files e.g., `readme.cast` is located in `../assets` directory relative to `example/index.html`.